### PR TITLE
feat:(composables): Use `useDevice` for device detecting.

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-8123

### CODE REVIEW

File: src/tailwind/plugin-options.js [hunk 1]
```diff
@@ -1,4 +1,11 @@
 /** @type {import('@empathyco/x-tailwindcss').PluginOptions} */
 module.exports = {
-  theme: {}
+  theme: {
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply src/tailwind/plugin-options.js]: done

File: src/components/search/related-tags.vue [hunk 2]
```diff
@@ -4,14 +4,13 @@
     class="x-sliding-panel-show-buttons-on-hover"
     buttonClass="x-button-lead x-button-circle x-button-ghost x-padding--00"
     scrollContainerClass="desktop:x-sliding-panel-fade"
-    :showButtons="$x.device !== 'mobile'"
+    :showButtons="isTouchable"
```
Comment (tiborux) [inline src/components/search/related-tags.vue:7]:
```suggestion
    :showButtons="!isTouchable"
```

File: src/components/my-history/custom-my-history.vue
Comment (tiborux) [inline src/components/my-history/custom-my-history.vue:1]: If you remove one history query, an horizonal scroll appears when the animation is done

File: src/components/search/results/results.vue
[the remark this answers is not in the document]
Comment (tajespasarela) [reply src/components/search/results/results.vue:1]: changed

File: src/components/my-history/custom-my-history.vue
Comment (tajespasarela) [reply src/components/my-history/custom-my-history.vue:1]: fixed

File: src/components/search/results/custom-next-query-preview.vue [hunk 3]
```diff
@@ -18,41 +18,51 @@
...
+      NextQueryPreview
+    },
+    props: {
+      nextQuery: { type: Object as PropType<NextQueryModel>, required: true }
+    },
+    setup(props) {
+      const { isTabletOrLess } = useDevice();
+      return {
+        ...toRefs(props),
+        maxItemsToRender: computed(() => (isTabletOrLess.value ? undefined : 5))
```
Comment (ajperezbau) [inline src/components/search/results/custom-next-query-preview.vue:62]: This is just a doubt about coding styles.
In the majority of the code written in Vue 3 I have seen, the things to return in the `setup` function are never declared here.
I suppose that it is that way to make the return easier to read, should we avoid declaring things here?

**Review by ajperezbau — CHANGES_REQUESTED** [review]
Good job! 👌

File: src/components/search/results/custom-next-query-preview.vue (hunk 3 above)
Comment (tiborux) [reply src/components/search/results/custom-next-query-preview.vue:62]: I think so

**Review by tiborux — APPROVED** [review]

**Review by ajperezbau — APPROVED** [review]

**Pull request merged**